### PR TITLE
feat(unity/mlmodel): add model signature and run details support in Unity Catalog Source

### PR DESCRIPTION
**Pull request by askumar27**
## Overview

This PR enhances the Unity Catalog ingestion to integrate MLflow run details (signatures, training metrics, and hyperparameters) into ML model metadata. The changes include comprehensive test coverage, type safety improvements, and robust error handling for production use.

### MLflow Run Details Integration

**Enhanced ML Model Metadata:**
- Added support for extracting comprehensive MLflow run details from Unity Catalog model versions
- Integrated model signatures with input/output schemas including types, dtypes, and shapes
- Extracted training metrics (e.g., accuracy, f1_score, precision) from MLflow runs
- Extracted hyperparameters (e.g., learning_rate, batch_size, epochs) from MLflow runs
- Captured MLflow tags (e.g., mlflow.user, mlflow.source.type, model_type)
- All metadata is now properly populated in DataHub MLModel entities

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/unity/source.py [hunk 1]
```diff
@@ -741,6 +743,16 @@ def process_ml_model_version(
                     created=TimeStampClass(time=created_time, actor=created_actor),
                 )
             )
+        custom_properties = {}
+        if ml_model_version.run_details:
+            if ml_model_version.run_details.signature:
+                custom_properties["signature"] = json.dumps(
```
Comment (yoonhyejin) [inline metadata-ingestion/src/datahub/ingestion/source/unity/source.py]: how is this represented in UI with different signature formats? is it just one custom properties - "signature"?

File: metadata-ingestion/src/datahub/ingestion/source/unity/proxy_types.py [hunk 2]
```diff
@@ -339,8 +339,77 @@ def add_upstream(cls, upstream: TableReference, notebook: "Notebook") -> "Notebo
...
+    run_id: str
+    experiment_id: str
+    status: Optional[str]
+    start_time: Optional[datetime]
+    end_time: Optional[datetime]
+    user_id: Optional[str]
+    metrics: Optional[Dict[str, str]]
+    parameters: Optional[Dict[str, str]]
+    tags: Optional[Dict[str, str]]
+    signature: Optional["ModelSignature"]
```
Comment (yoonhyejin) [inline metadata-ingestion/src/datahub/ingestion/source/unity/proxy_types.py]: IMO signature should be a part of ModelVersion since this is more directly related to each model version & we extract this from modelversion API

File: metadata-ingestion/src/datahub/ingestion/source/unity/proxy.py [hunk 3]
```diff
@@ -187,6 +199,264 @@ def __init__(
...
+
+            # Extract tags
+            tags: Dict[str, str] = {}
+            if run.data and run.data.tags:
+                for tag in run.data.tags:
+                    if tag.key is not None and tag.value is not None:
+                        tags[tag.key] = tag.value
+
+            # Extract signature using Files API if model_version is provided
+            signature: Optional[ModelSignature] = None
```
Comment (yoonhyejin) [inline metadata-ingestion/src/datahub/ingestion/source/unity/proxy.py]: IMO this should be separated from fetching runs info - we're using a different endpoint, this is not directly related to run details

File: metadata-ingestion/src/datahub/ingestion/source/unity/source.py (hunk 1 above)
Comment (yoonhyejin) [reply metadata-ingestion/src/datahub/ingestion/source/unity/source.py]: Without the UI change - what we can do very quickly is separate signature - input, signature - output, signature - parameters for different keys when we emit it to custom properties, rather than have it in one signature

and we can trim the json a little bit prettier -- maybe at least with line breaks

Comment (askumar27) [reply metadata-ingestion/src/datahub/ingestion/source/unity/source.py]: Updated to generate:
```
 "customProperties": {
                "signature.inputs": "[{\"type\": \"tensor\", \"tensor-spec\": {\"dtype\": \"float64\", \"shape\": [-1, 10]}}]",
                "signature.outputs": "[{\"type\": \"tensor\", \"tensor-spec\": {\"dtype\": \"int64\", \"shape\": [-1]}}]",
                "mlflow.runColor": "\"#da4c4c\"",
                "mlflow.runName": "\"unity_catalog_demo\"",
                "mlflow.source.name": "\"mlflow_with_unity.py\"",
                "mlflow.source.type": "\"LOCAL\"",
                "mlflow.user": "\"test@test.io\"",
                "mlflow.artifacts.spn": "\"/WorkspaceInternal/Mlflow/Artifacts/1675014323606054/Runs/a68c6bd4e9e044539231f9e838025751/\""
            },
```

File: metadata-ingestion/src/datahub/ingestion/source/unity/proxy_types.py (hunk 2 above)
Comment (askumar27) [reply metadata-ingestion/src/datahub/ingestion/source/unity/proxy_types.py]: Moved it under `ModelVersion`

File: metadata-ingestion/src/datahub/ingestion/source/unity/proxy.py (hunk 3 above)
Comment (askumar27) [reply metadata-ingestion/src/datahub/ingestion/source/unity/proxy.py]: decoupled it from `get_run_details` method - independently populated for `ModelVersion`

**Review by yoonhyejin — APPROVED** [review]
LGTM - lint and codecov is still failing, can you take a look at it?
